docs/usage/shoot_pod_autoscaling_best_practices.md

```diff
@@ -0,0 +1,337 @@
+
+Finally, the behavior section influences how fast you scale up and down. Most of the time (depends on your workload), you like to scale out faster than you scale in. In this example, the configuration will trigger a scale-out only after observing the need to scale out for 30s (`stabilizationWindowSeconds`) and will then only scale out at most 100% (`value` + `type`) of the current number of replicas every 60s (`periodSeconds`). The configuration will trigger a scale-in only after observing the need to scale in for 1800s (`stabilizationWindowSeconds`) and will then only scale in at most 1 pod (`value` + `type`) every 300s (`periodSeconds`). As you can see, scale-out happens quicker than scale-in in this example.
+
+## HPA (actually KCM) Options
+
+HPA is a function of the kube-controller-manager (KCM).
+
+You can read up the [full KCM options](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager) online and set most of them conveniently in your [Gardener shoot cluster spec](https://github.com/gardener/gardener/blob/957e4c7/example/90-shoot.yaml#L221-L226):
+- `downscaleStabilization` (default 5m): HPA will scale out whenever the formula (in accordance with the behavior section, if present in the HPA resource) yields a higher replica count, but it won't scale in just as eagerly. This option lets you define a trailing time window that HPA must check and only if the recommended replica count is consistently lower throughout the entire time window, HPA will scale in (in accordance with the behavior section, if present in the HPA resource). If at any point in time in that trailing time window the recommended replica count isn't lower, scale-in won't happen. This setting is just a default, if nothing is defined in the behavior section of an HPA resource. The default for the upscale stabilization is 0s and it cannot be set via a KCM option ([downscale stabilization was historically more important than upscale stabilization and when later the behavior sections were added to the HPA resources, upscale stabilization remained missing from the KCM options](https://github.com/kubernetes/enhancements/blob/master/keps/sig-autoscaling/853-configurable-hpa-scale-velocity/README.md)).
+- `tolerance` (default +/-10%): HPA will not scale out or in if the desired replica count is (mathematically as a float) near the actual replica count (see [source code](https://github.com/kubernetes/kubernetes/blob/f3f5dd99ac7bdc61c61c3d587575090c3473ab5a/pkg/controller/podautoscaler/replica_calculator.go#L97-L103) for details), which is a form of hysteresis to avoid replica flapping around a threshold.
+
+There are a few more configurable options of lesser interest:
```

Comment:
Lesser interest to the general public, but maybe of interest to some end users.

The reason I was explaining those is that they are configurable in Gardener (I am not sure why only those exactly, but probably we needed them at some point in time, which speaks for "some basic value").

Also, I wanted to avoid misunderstandings, which is why I researched (here) `cpuInitializationPeriod` and `initialReadinessDelay` in code, because I (and probably also other/most readers) thought this works differently and could be a solution to black-out the high CPU peak load when a component starts up when in fact, it's not that trivial.